Engine/Helper.cs

```diff
@@ -715,67 +716,24 @@ private CommandInfo GetCommandInfoInternal(string cmdName, CommandTypes? command
         }
 
         /// <summary>
-
-        ///  Legacy method, new callers should use <see cref="GetCommandInfo"/> instead.
-        ///  Given a command's name, checks whether it exists. It does not use the passed in CommandTypes parameter, which is a bug.
-        ///  But existing method callers are already depending on this behaviour and therefore this could not be simply fixed.
-        ///  It also populates the commandInfoCache which can have side effects in some cases.
+        /// Get a CommandInfo object of the given command name and type
+        /// <param name="name"></param>
+        /// <param name="commandType"></param>
         /// </summary>
-        /// <param name="name">Command Name.</param>
-        /// <param name="commandType">Not being used.</param>
-        /// <returns></returns>
-        [Obsolete]
-        public CommandInfo GetCommandInfoLegacy(string name, CommandTypes? commandType = null)
+        /// <returns>The first CommandInfo found based on the name and type or null if nothing was found</returns>
+        public CommandInfo GetCommandInfo(string name, CommandTypes commandType)
         {
-            if (string.IsNullOrWhiteSpace(name))
-            {
-                return null;
-            }
-
-            // check if it is an alias
-            string cmdletName = Helper.Instance.GetCmdletNameFromAlias(name);
-            if (string.IsNullOrWhiteSpace(cmdletName))
-            {
-                cmdletName = name;
-            }
-
-            lock (getCommandLock)
-            {
-                if (commandInfoCache.ContainsKey(cmdletName))
-                {
-                    return commandInfoCache[cmdletName];
-                }
-
-                var commandInfo = GetCommandInfoInternal(cmdletName, commandType);
-                commandInfoCache.Add(cmdletName, commandInfo);
```

Comment:
If we do not add to the `commandInfoCache` any more (not sure why we can remove it or why it was there in the first place?), then we should remove this variable entirely since there do not seem to be any other references to it.

Reply:
that is a big error - I meant to leave it in the Internal routine (where it belongs) - updated PR on the way

Reply:
OK. I am interested to see how you get along with that. I remember when I tried to do something similar, it broke 10 other unrelated tests, hence why I ended up creating a new function that does not add to the `commandInfoCache`. This and most other stuff in the Helper class seems to be SSA related and unfortunately the current SSA implementation is far away from being complete/usable (this is the main reason why the PR on warning on `+=` for arrays will probably be in WIP state for a while).
Going forward in future PRs, what do you think is the best way to deal with this Helper class that has so many code smells like static members and is probably the culprit for a few issues that PSSA has around exceptions being thrown due to each rules being executed in a multi-threaded manner? Should we try to avoid touching it and write new stuff in a new place and where possible move code over to the new/better place? I have the feeling that the boy-scout rule is very hard to apply to this class.